In [ ]:
# Graded 1
# 3.1.6.9
# list set dict auto remove duplicate value

def letter_catalog(items,letter='A'):
  temp_list = []
  for i in range(len(items)):
    if items[i].startswith(letter) == True:
      temp_list.append(items[i])

  return temp_list
  
# Cek output kode anda
letter_catalog(['Apple','Avocado','Banana','Blackberries','Blueberries','Cherries'],letter='A')



['Apple', 'Avocado']

In [ ]:
# Graded 2
# 4.1.6.8

def counter_item(items):
  value = []
  for i in items: 
    value.append(items.count(i))

  new_dict = {}
  values = []
  for i in range(len(items)):
   if (items[i] not in new_dict) and (value[i] not in values):
     new_dict.update({items[i]: value[i]})

  return new_dict

# Cek output kode anda
counter_item(['Apple','Apple','Apple','Blueberries','Blueberries','Blueberries'])


{'Apple': 3, 'Blueberries': 3}

In [ ]:
# Graded 3
# 4.1.6.6

# dua variable berikut jangan diubah
fruits = ['Apple','Avocado','Banana','Blackberries','Blueberries','Cherries','Date Fruit','Grapes','Guava','Jackfruit','Kiwifruit']
prices = [6,5,3,10,12,7,14,15,8,7,9]

# list buah
chart = ['Blueberries','Blueberries','Grapes','Apple','Apple','Apple','Blueberries','Guava','Jackfruit','Blueberries','Jackfruit']

fruit_price = None
fruit_price = {}
for i in range(len(fruits)):
  fruit_price.update({fruits[i]: prices[i]})

def total_price(dcounter,fprice):
  buy_fprice = [] 
  for i in list(dcounter.keys()):
    if i in fprice:
      buy_fprice.append(fprice[i])

  all_price = 0
  for i in range(len(buy_fprice)):
    all_price += list(dcounter.values())[i] * buy_fprice[i]

  return all_price

# Cek output kode anda
total_price(counter_item(chart),fruit_price)


103

In [ ]:
# Graded 4
# 3.1.7.1

def discounted_price(total,discount,minprice=100):
  disc_price = 0
  if total >= minprice:
    disc_price = total - total * discount / 100
  else:
    disc_price = total
  return disc_price

# Cek output kode anda
discounted_price(total_price(counter_item(chart),fruit_price),10,minprice=100)


92.7

In [ ]:
# Graded 5
# 4.1.6.7

def print_summary(items,fprice):
  new_items = counter_item(items)
  sort_items = sorted(new_items)
  sort_dict = dict(sorted(new_items.items()))
  buy_fprice = [] 
  for i in sort_items:
    if i in fprice:
      buy_fprice.append(fprice[i])

  for i in range(len(new_items)):
    print(list(sort_dict.values())[i], list(sort_dict.keys())[i], ":", list(sort_dict.values())[i] * buy_fprice[i])
    continue
    break

  print("total", ":", total_price(new_items, fprice))
  print("discount price", ":", discounted_price(total_price(new_items, fprice), 10, minprice=100))

# Cek output kode anda
print_summary(chart,fruit_price)

# Expected

3 Apple : 18

4 Blueberries : 48

1 Grapes : 15

1 Guava : 8

2 Jackfruit : 14

total : 103

discount price : 92.7